### PR TITLE
Fetch all addons for settings and panel

### PR DESCRIPTION
**Pull request by danirabbit**
This branch fetches all addons for the panel and system settings instead of hardcoding a list to load.

Depends on:
- [x] https://github.com/elementary/appstream-data/pull/29

### CODE REVIEW

File: src/MainWindow.vala [hunk 1]
```diff
@@ -215,6 +215,18 @@ public class Feedback.MainWindow : Gtk.ApplicationWindow {
         });
     }
 
+    private GLib.Icon get_extension_icon_from_appstream (GLib.GenericArray<AppStream.Icon> appstream_icons) {
+        if (appstream_icons.length > 0) {
```
Comment (tintou) [inline src/MainWindow.vala:219]: you don't need this

File: src/MainWindow.vala [hunk 2]
```diff
@@ -215,6 +215,18 @@ public class Feedback.MainWindow : Gtk.ApplicationWindow {
...
+            for (int i = 0; i < appstream_icons.length; i++) {
```
Comment (tintou) [inline src/MainWindow.vala:220]:
```suggestion
            foreach (unowned AppStream.Icon appstream_icon in appstream_icons) {
```
This works too and is more vala-esque

Comment (danrabbit) [reply src/MainWindow.vala:220]: For this I get `GLib.GenericArray<AppStream.Icon>' does not have an 'iterator'` which is why I did the other way

Comment (tintou) [reply src/MainWindow.vala:220]: Ah indeed, opening a PR for Vala https://gitlab.gnome.org/GNOME/vala/-/merge_requests/146

**Review by zeebok — APPROVED** [review]

**Pull request merged**